### DISCUSSION TITLE
New Test- and Get- commands for Availability Groups

### DISCUSSION THREAD

**Discussion by andreasjordan** [dataplat/dbatools#7693] [Contributors]
Some months ago I scanned all issues for interesting issues I wanted to work on. I worked on most of them, but now I have some that I can't work on alone. I still find then interesting so I will link them here and maybe someone else wants to join and work on them together:

#2190: Test-DbaAvailabilityGroup (I already created the command, but basically only as a test for Add-DbaAgDatabase.)
#3806: New Command: Test-DbaAgHealth
#3807: New Command: Get-DbaAgHealth
#3808: New command: Get-DbaAgReplicaState

**Comment by tboggiano** [comment]
@andreasjordan I'd like to take some of these on, these are all things I can use as part of automating my patching processes at work.

**Reply by andreasjordan** [reply]
Great! I would be happy to do a review and give feedback.

**Reply by kjustin765** [reply]
@andreasjordan Still need some help on any of these?

**Reply by andreasjordan** [reply]
Yes, help is welcome. I currently don't work with Availability Groups as a dba, so I don't have a good environment to test the internals of those commands. So if you have ideas on what and how to test the health of AGs, then you are more than welcome to share your ideas and code.
